sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -909,28 +912,44 @@ int gcoap_get_resource_list(void *buf, size_t maxlen, uint8_t cf)
     char *out = (char *)buf;
     size_t pos = 0;
 
+    coap_link_encoder_ctx_t ctx;
+    ctx.content_format = cf;
+
     /* write payload */
     while (listener) {
+        if (!listener->link_encoder) {
+            continue;
+        }
         const coap_resource_t *resource = listener->resources;
+        ctx.link_pos = 0;
 
-        for (unsigned i = 0; i < listener->resources_len; i++) {
-            size_t path_len = strlen(resource->path);
+        for (; ctx.link_pos < listener->resources_len; ctx.link_pos++) {
             if (out) {
-                /* only add new resources if there is space in the buffer */
-                if ((pos + path_len + 3) > maxlen) {
+                if (pos + 1 > maxlen) {
                     break;
                 }
                 if (pos) {
                     out[pos++] = ',';
```

Comment:
This is Link format specific, I know that right now this function only accepts Link Format. but why can't this be moved to `gcoap_encode_link`, I believe that the position in the context was meant for this right?

Reply:
There are advantages to writing the separator in either place, and I don't feel too strongly about it. Actually, the `link_pos` attribute of the context is not sufficient to identify when to include the separator. link_pos is the position with a listener, not the position within the overall list. (I clarified the documentation on this meaning.)

So, I added a `flags ` attribute that includes when to write the link separator. More in a separate comment.